# HG changeset patch
# User Iain Murray  <iain@iainmurray.net>
# Date 1300312809 25200
#      Wed Mar 16 15:00:09 2011 -0700
# Node ID a1b2da4967ac7aa19a62e3deb2e835d3853cd99d
# Parent  e742720c5e71c72d156ac672d0572542388ee628
Add nproc, nproc_conf utility functions for determining number of available processors.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2010-03-16  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/system.txi: Add nproc and nproc_conf functions to
+	documentation.
+
 2010-03-03  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/matrix.txi: Deprecate is_duplicate_entry.
 
 2010-03-02  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/stats.txi: Correctly refer to discrete uniform 
 	distribution in documentation. 
diff --git a/doc/interpreter/system.txi b/doc/interpreter/system.txi
--- a/doc/interpreter/system.txi
+++ b/doc/interpreter/system.txi
@@ -465,16 +465,20 @@ referred to as a @var{grp_struct}.
 
 @node System Information
 @section System Information
 
 @DOCSTRING(computer)
 
 @DOCSTRING(uname)
 
+@DOCSTRING(nprocs)
+
+@DOCSTRING(nprocs_conf)
+
 @DOCSTRING(ispc)
 
 @DOCSTRING(isunix)
 
 @DOCSTRING(ismac)
 
 @DOCSTRING(isieee)
 
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2011-03-16  Iain Murray  <iain@iainmurray.net>
+
+	* DLD-FUNCTIONS/nprocs.cc: New file.
+	* DLD-FUNCTIONS/module-files: Add nprocs.cc.
+	Expose nprocs and nprocs_conf from gnulib.
+
 2010-03-15  Marco Caliari  <marco.caliari@univr.it>
 
 	* graphics.cc: Simplify calculation of number of tick labels.  Fixes
 	bug #32692.
 
 2011-03-13  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* DLD-FUNCTIONS/__init_fltk__.cc (plot_window::plot_window):
diff --git a/src/DLD-FUNCTIONS/module-files b/src/DLD-FUNCTIONS/module-files
--- a/src/DLD-FUNCTIONS/module-files
+++ b/src/DLD-FUNCTIONS/module-files
@@ -50,16 +50,17 @@ kron.cc
 lookup.cc
 lsode.cc
 lu.cc
 luinc.cc
 matrix_type.cc
 max.cc
 md5sum.cc
 mgorth.cc
+nprocs.cc
 onCleanup.cc
 pinv.cc
 qr.cc
 quad.cc
 quadcc.cc
 qz.cc
 rand.cc
 rcond.cc
diff --git a/src/DLD-FUNCTIONS/nprocs.cc b/src/DLD-FUNCTIONS/nprocs.cc
new file mode 100644
--- /dev/null
+++ b/src/DLD-FUNCTIONS/nprocs.cc
@@ -0,0 +1,76 @@
+/*
+
+Copyright (C) 2011 Iain Murray
+
+
+This file is part of Octave.
+
+Octave is free software; you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 3 of the License, or (at your
+option) any later version.
+
+Octave is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with Octave; see the file COPYING.  If not, see
+<http://www.gnu.org/licenses/>.
+
+*/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "defun-dld.h"
+#include "sys/sysinfo.h"
+
+DEFUN_DLD (nprocs, args, nargout,
+   "-*- texinfo -*-\n\
+@deftypefn {Loadable Function} {} nprocs ()\n\
+Return the number of available processors.\n\
+@seealso{nprocs_conf}\n\
+@end deftypefn")
+{
+  octave_value retval;
+
+  int nargin = args.length ();
+
+  if (nargin != 0 || (nargout != 0 && nargout != 1))
+    {
+      print_usage ();
+      return retval;
+    }
+
+  retval = get_nprocs ();
+
+  return retval;
+}
+
+DEFUN_DLD (nprocs_conf, args, nargout,
+   "-*- texinfo -*-\n\
+@deftypefn {Loadable Function} {} nprocs_conf ()\n\
+Return the number of number of processors the operating system has\n\
+configured.  This number may be less than the total available as reported by\n\
+@code{nprocs}.\n\
+@seealso{nprocs}\n\
+@end deftypefn")
+{
+  octave_value retval;
+
+  int nargin = args.length ();
+
+  if (nargin != 0 || (nargout != 0 && nargout != 1))
+    {
+      print_usage ();
+      return retval;
+    }
+
+  retval = get_nprocs_conf ();
+
+  return retval;
+}
+
